src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,136 @@
+package parser;
+
+import exception.InvalidArgumentException;
+import exception.MissingArgumentException;
+import response.CreateResponse;
+import response.DeadlineResponse;
+import response.DeleteResponse;
+import response.EventResponse;
+import response.FindResponse;
+import response.ListResponse;
+import response.MarkResponse;
+import response.Response;
+import response.UnMarkResponse;
+
+/**
+ * Represents the parser for the user input
+ */
+public class Parser {
+
+    private enum Types { LIST, MARK, UNMARK, TODO, DEADLINE, EVENT, DELETE, FIND }
+
+    /**
+     * Represents the user input
+     */
+    private String input;
+    private String inputType;
+    private String inputContent;
+
+
+    /**
+     * Creates a parser for the specified user input
+     * @param input User input
+     */
+    public Parser(String input) {
+        this.input = input;
+    }
+
+    /**
+     * Parses the user input
+     * @return Response returned, depending on the user input parsed
+     */
+    public Response parse() throws MissingArgumentException, InvalidArgumentException {
+        categorisation();
+        switch (this.inputType) {
+        case "LIST":
+            return new ListResponse();
+        case "MARK":
+            return new MarkResponse(this.inputContent);
+        case "UNMARK":
+            return new UnMarkResponse(this.inputContent);
+        case "TODO":
+            return new CreateResponse(this.inputContent);
+        case "DEADLINE":
+            return new DeadlineResponse(this.inputContent);
+        case "EVENT":
+            return new EventResponse(this.inputContent);
+        case "DELETE":
+            return new DeleteResponse(this.inputContent);
+        case "FIND":
+            return new FindResponse(this.inputContent);
+        default:
+            throw new MissingArgumentException(
+                    "I'm sorry, but I don't know what that means :-("
+                            + "\n\t  To create a new todo, use 'todo ...',"
+                            + "\n\t  To create a new event, use 'event ... /from ... /to ...',"
+                            + "\n\t  To create a new deadline, use 'deadline ... /by ...',"
+                            + "\n\t  To list all tasks, use 'list',"
+                            + "\n\t  To mark a task, use 'mark ' with a number,"
+                            + "\n\t  To unmark a task, use 'unmark ' with a number,"
+                            + "\n\t  Finally to exit the program, use 'bye'!");
+        }
+    }
+
+    /**
+     * Normalises all the whitespaces in a string
+     * @param s Input string to normalise
+     * @return The normalised string
+     */
+    public String normaliseAllWhitespaces(String s) {
+        int l = s.length();
+        StringBuilder sb = new StringBuilder(l);
+
+        boolean lastWasWhite = false;
+        boolean modified = false;
```

Comment:
Perhaps better naming can be used for boolean variables?
```suggestion
        boolean wasLastWhite = false;
        boolean isModified = false;
```